app/src/features/inntektsmelding/Steg3Oppsummering.tsx

```diff
@@ -72,42 +73,33 @@ type SendInnInntektsmeldingProps = {
 function SendInnInntektsmelding({ opplysninger }: SendInnInntektsmeldingProps) {
   const navigate = useNavigate();
   const { id } = route.useParams();
+  const { eksisterendeInntektsmeldinger } = route.useLoaderData();
 
   const { gyldigInntektsmeldingSkjemaState, setInntektsmeldingSkjemaState } =
     useInntektsmeldingSkjema();
 
   const { mutate, error, isPending } = useMutation({
     mutationFn: async (skjemaState: InntektsmeldingSkjemaStateValid) => {
-      const gjeldendeInntekt =
-        skjemaState.korrigertInntekt ?? skjemaState.inntekt;
-
-      const refusjon =
-        skjemaState.skalRefunderes === "JA_LIK_REFUSJON"
-          ? skjemaState.refusjon.slice(0, 1)
-          : skjemaState.skalRefunderes === "JA_VARIERENDE_REFUSJON"
-            ? skjemaState.refusjon
-            : [];
-
-      const inntektsmelding = {
-        foresporselUuid: id,
-        aktorId: opplysninger.person.aktørId,
-        ytelse: opplysninger.ytelse,
-        arbeidsgiverIdent: opplysninger.arbeidsgiver.organisasjonNummer,
-        kontaktperson: skjemaState.kontaktperson,
-        startdato: opplysninger.startdatoPermisjon,
-        inntekt: formatStrengTilTall(gjeldendeInntekt),
-        // inntektEndringsÅrsak: skjemaState.inntektEndringsÅrsak, // Send inn når BE har støtte for det
-        refusjon: refusjon.map((r) => ({
-          ...r,
-          beløp: formatStrengTilTall(r.beløp),
-        })),
-        bortfaltNaturalytelsePerioder: konverterNaturalytelsePerioder(
-          skjemaState.naturalytelserSomMistes,
-        ),
-        endringAvInntektÅrsaker: skjemaState.endringAvInntektÅrsaker,
-      } satisfies SendInntektsmeldingRequestDto;
-
-      return sendInntektsmelding(inntektsmelding);
+      const inntektsmeldingRequest = lagSendInntektsmeldingRequest(
+        id,
+        skjemaState,
+        opplysninger,
+      );
+
+      if (eksisterendeInntektsmeldinger[0]) {
+        const eksisterendeInntektsmelding = lagSendInntektsmeldingRequest(
+          id,
+          eksisterendeInntektsmeldinger[0],
+          opplysninger,
+        );
+        if (isEqual(inntektsmeldingRequest, eksisterendeInntektsmelding)) {
+          throw new Error(
+            "Du har ikke gjort noen endringer fra forrige innsendte inntektsmelding.",
+          );
+        }
```

Comment:
Istedenfor å bruke lodash her, kan du bruke `JSON.stringify(inntektsmeldingRequest) === JSON.stringify(eksisterendeInntektsmelding)` for å gjøre en kjapp og billig sjekk.

Gitt at alle props er på riktig plass, da.


```ts
function erLike(a: unknown, b: unknown) {
  return JSON.stringify(a) === JSON.stringify(b);
}
```

Reply:
I know, men jeg synes ikke noe om at rekkefølgen på properties skal ha betydning

Reply:
Så lenge man drar inn individuelle imports fra Lodash vil man jo slippe voldsomt større bundle size. Hvis det er [dette biblioteket](https://bundlephobia.com/package/deep-equal@2.2.3) du viste til, så er det faktisk større enn å importere `lodash/isEqual`

Reply:
Om det er samme rekkefølge i vårt tilfelle, så er det jo ikke noe problem. Men om det viser seg å trenges, så skal du aller nødigst få lov til å importere lodash.

<img src="https://media3.giphy.com/media/FYCxLHCIGaioo/giphy.gif"/>